tests/backup/backup_psa_test.go

```diff
@@ -943,3 +944,460 @@ var _ = Describe("{RestoreFromHigherPrivilegedNamespaceToLower}", Label(TestCase
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of restore with replace option %s from backup %s", restoreName, allNSScheduleBackup))
+		})
+
+		Step("Custom restore of restricted namespaces to destination cluster", func() {
+			log.InfoD("Custom restore of restricted namespaces to destination cluster")
+			restoreName := fmt.Sprintf("restore-restricted-%v", RandomString(5))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestoreWithProjectMapping(restoreName, restrictedScheduleBackup, destNamespaceProjectMapping, DestinationClusterName, BackupOrgID, ctx, nil, restoreProjectUIDMapping, restoreProjectNameMapping)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Custom restore %s from restricted namespace backup %s", restoreName, restrictedScheduleBackup))
+		})
+
+		Step("Default restore of baseline namespaces to destination cluster", func() {
+			log.InfoD("Default restore of baseline namespaces to destination cluster")
+			restoreName := fmt.Sprintf("restore-baseline-%v", RandomString(5))
+			restoreList = append(restoreList, restoreName)
+			err = CreateRestoreWithValidation(ctx, restoreName, baselineScheduleBackup, make(map[string]string), make(map[string]string), DestinationClusterName, BackupOrgID, baselineScheduledAppContexts)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Default restore %s from baseline namespace backup %s", restoreName, baselineScheduleBackup))
+		})
+
+	})
+
+	JustAfterEach(func() {
+		log.Infof("Cleaning up the resources after the test")
+		defer func() {
+			Inst().AppList = actualAppList
+		}()
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		defer func() {
+			log.Infof("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "Failed to SetClusterContext to default cluster")
+		}()
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		log.Infof("Deleting the backups as part of clean up")
+		for _, backup := range backupList {
+			backupUid, err := Inst().Backup.GetBackupUID(ctx, backup, BackupOrgID)
+			log.FailOnError(err, "Unable to fetch backup UID")
+			_, err = DeleteBackup(backup, backupUid, BackupOrgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting backup [%s]", backup))
+		}
+		log.Info("Destroying scheduled apps on source cluster")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		DestroyApps(scheduledAppContexts, opts)
+
+		log.Infof("Deleting projects from source cluster")
+		for i, project := range sourceClusterProjectList {
+			err = Inst().S.(*rke.Rancher).DeleteRancherProject(sourceClusterProjectUIDList[i])
+			log.FailOnError(err, "Deletion of project %s failed", project)
+		}
+		log.Info("Deleting restored namespaces")
+		for _, restoreName := range restoreList {
```

Comment:
can remove this code as it is taken care when cluster is deleted

Reply:
Done